### PR TITLE
refactor(triton): reorganize conv modules and unify gated FP8 quant path

### PR DESCRIPTION
**Pull request by hellozhuo-amd**
## Summary

This PR reorganizes Triton conv1d code into dedicated `conv` modules and folds gated RMS+FP8 quantization into the existing fused FP8 kernel path.
Related comments: https://github.com/ROCm/aiter/pull/3005#issuecomment-4372179376
Original PR (pre-organized): https://github.com/ROCm/aiter/pull/2423, merged at https://github.com/ROCm/aiter/pull/3005.

### What changed

- **Conv module reorganization**
  - Moved conv1d Triton kernels under:
    - `aiter/ops/triton/_triton_kernels/conv/`
  - Moved conv1d Python wrappers under:
    - `aiter/ops/triton/conv/`
  - Add alias at `aiter/ops/triton/__init__.py` to enable the equivalence of the following import styles for compatibility:
 ```python
from aiter.ops.triton.causal_conv1d[__update_single_token]
from aiter.ops.triton.conv.causal_conv1d[__update_single_token]
```

- **FP8 quant integration**
  - Unified gated RMS+FP8 flow as a feature path in the existing fused FP8 implementation instead of maintaining a separate standalone wrapper path.
  - Kept both:
    - classic path (`GATED_RMS_FP8=False`), and
    - gated path (`GATED_RMS_FP8=True`),
    in the same fused kernel flow.
  - Preserved gated launch behavior with dynamic `ROWS_PER_BLOCK` (`calc_rows_per_block`) for gated execution.

- **Test layout + script updates**
  - Moved conv tests to:
    - `op_tests/triton_tests/conv/test_causal_conv1d.py`
    - `op_tests/triton_tests/conv/test_causal_conv1d_update_single_token.py`
  - Updated split test mapping:
    - `.github/scripts/split_tests.sh`

## Motivation

- Align conv1d structure with existing Triton package organization patterns (`attention`, `quant`, etc.) for easier discovery and maintenance.
- Reduce duplicate code paths in FP8 quantization by treating gating as an additional feature on top of the existing fused kernel path.
- Keep test coverage explicit for both gated and classic quant paths after refactor.

## Technical Details

- Introduced new conv package roots in both kernel and wrapper layers.
- Updated conv test paths accordingly.
- Consolidated gated/classic quant control flow under the fused FP8 kernel path with feature flags.
- Retained behavior-sensitive launch configuration for gated quant via dynamic rows-per-block heuristic.

## Test Plan

Ran:

```bash
python3 -m pytest \
  op_tests/triton_tests/test_fused_rearrange_sigmoid_gdr.py \
  op_tests/triton_tests/conv/test_causal_conv1d_update_single_token.py \
  op_tests/triton_tests/quant/test_fused_rms_gated_fp8_group_quant.py \
  op_tests/triton_tests/quant/test_fused_fp8_quant.py::test_fused_rms_fp8_group_quant \
  -v
```

Additional validation:

- `black --check` and `ruff check` on touched Triton conv/quant files.

## Test Result

- Targeted test bundle passed.
- Gated quant sweep + classic fused FP8 quant tests passed.
- Formatting/lint checks on touched files passed.

From the vllm side, the speed of the gated fp8 kernel was maintained.

## Notes

- No functional regressions observed in covered paths.
- Includes path/layout refactor, so downstream imports should use the new `conv` module paths.

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

### CODE REVIEW

File: aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py [hunk 1]
```diff
@@ -181,88 +181,215 @@ def _fused_rms_fp8_group_quant_kernel(
...
+            off_rms += RMS_TILE
+
+        var = sumsq / N
+        rstd = tl.math.rsqrt(var + eps)
+
+        for g in range(NUM_GROUPS_GATED):
+            col0 = g * GROUP_SIZE_GATED
+            cols = tl.arange(0, BLOCK_G) + col0
+            col_mask = cols < N
+            mask_g = row_mask_1d[:, None] & col_mask[None, :]
```
[the remark this answers is not in the document]
Comment (hellozhuo-amd) [reply aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:266]: @copilot apply changes based on [this feedback](https://github.com/ROCm/aiter/pull/3048#discussion_r3194166433)

File: aiter/ops/triton/quant/fused_fp8_quant.py [hunk 2]
```diff
@@ -283,6 +282,9 @@ def fused_rms_fp8_group_quant(
         out1_bs_row_stride = out1_bs.stride(0)
         out1_bs_col_stride = out1_bs.stride(1)
 
+    num_groups_g = num_bs_cols
+    block_g_gated = triton.next_power_of_2(group_size)
+
```
[the remark this answers is not in the document]
Comment (hellozhuo-amd) [reply aiter/ops/triton/quant/fused_fp8_quant.py]: @copilot apply changes based on [this feedback](https://github.com/ROCm/aiter/pull/3048#discussion_r3194166474) Particularly, passing constant placeholders for the gated-only constexprs in the non-gated launch.

File: aiter/ops/triton/_triton_kernels/conv/__init__.py
Comment (Boss2002n) [inline aiter/ops/triton/_triton_kernels/conv/__init__.py:1]: can we keep this file empty

File: aiter/ops/triton/conv/__init__.py
Comment (Boss2002n) [inline aiter/ops/triton/conv/__init__.py:1]: we can empty this file too, unless theres a reason as to why we need this. 
It will be consistent with the rest of the repo if we empty it :)

File: aiter/ops/triton/quant/fused_fp8_quant.py [hunk 3]
```diff
@@ -461,34 +445,71 @@ def fused_rms_gated_fp8_group_quant(
...
-        eps,
+        1,
+        1,
+        1,
+        1,
+        1,
+        1,
+        z,
+        bias_ptr,
+        z.stride(0),
```
Comment (Boss2002n) [inline aiter/ops/triton/quant/fused_fp8_quant.py]: can we not do positional args for this, keyword args make it easy to read, it is very hard to follow given we have so many args being passed

File: aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py
Comment (Boss2002n) [inline aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:1]: I am wondering if we can split these into 2 different kernels instead of having an if-else.
would help making it easier to read

**Review by Boss2002n — CHANGES_REQUESTED** [review]

File: aiter/ops/triton/_triton_kernels/conv/__init__.py
Comment (hellozhuo-amd) [reply aiter/ops/triton/_triton_kernels/conv/__init__.py:1]: Thanks, it's empty now.

File: aiter/ops/triton/conv/__init__.py
Comment (hellozhuo-amd) [reply aiter/ops/triton/conv/__init__.py:1]: Thanks, yeah it is empty now

File: aiter/ops/triton/quant/fused_fp8_quant.py (hunk 3 above)
Comment (hellozhuo-amd) [reply aiter/ops/triton/quant/fused_fp8_quant.py]: Sure, positional args added.

File: aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py
Comment (hellozhuo-amd) [reply aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:1]: The integration into a single kernel (the unified kernel) is suggested by Ali Zaidy @azaidy :
"the fp8 quant can be integrated into existing ones (I think). its an additional feature on top of existing ones. so maybe if you add the feature into existing kernels its lesser code to maintain."

For the speed of the unified kernel, I was worried about the introduction of additional constexpr parameters could affect the original speed, but it turned out it was good as tested from the vllm side.

Comment (k50112113) [reply aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:1]: @hellozhuo-amd can you do a quick rocprof type of profiling to see if the performance is affected? I think at least we would like to make sure the launch-bound shapes are still launch-bound, for example M = 64, N = 7168

Comment (k50112113) [reply aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:1]: as of right now, the code looks good, but just please make sure the performance is not affected using rocprof

Comment (hellozhuo-amd) [reply aiter/ops/triton/_triton_kernels/quant/fused_fp8_quant.py:1]: @k50112113 @Boss2002n  Had a rocprof profiling for both functions (main is the baseline without pr), no regression found at the moment:
| Function | M | N | BLOCK_SIZE_N | Branch | Avg(us) | Min(us) | Max(us) | VGPR | SGPR |
|---|---|---|---|---|---|---|---|---|---|
| `fused_rms_fp8_group_quant`       |   64 | 7168 | 8192 | main |  5.2 |  5.1 |  5.4 | 16 | 48 |
| `fused_rms_fp8_group_quant`       |   64 | 7168 | 8192 | PR   |  5.2 |  4.8 |  5.5 | 16 | 48 |
| `fused_rms_fp8_group_quant`       |  512 | 7168 | 8192 | main |  6.1 |  6.0 |  6.2 | 16 | 48 |
| `fused_rms_fp8_group_quant`       |  512 | 7168 | 8192 | PR   |  6.3 |  6.3 |  6.3 | 16 | 48 |
| `fused_rms_fp8_group_quant`       | 4096 | 7168 | 8192 | main | 24.9 | 24.0 | 25.7 | 16 | 48 |
| `fused_rms_fp8_group_quant`       | 4096 | 7168 | 8192 | PR   | 24.9 | 24.2 | 25.6 | 16 | 48 |
| `fused_rms_fp8_group_quant`       |   64 | 4096 | 4096 | main |  4.7 |  4.1 |  5.2 | 12 | 48 |
| `fused_rms_fp8_group_quant`       |   64 | 4096 | 4096 | PR   |  4.3 |  3.5 |  5.0 | 12 | 48 |
| `fused_rms_gated_fp8_group_quant` |   64 | 7168 | 8192 | main |   —  |   —  |   —  |  — |  — |
| `fused_rms_gated_fp8_group_quant` |   64 | 7168 | 8192 | PR   | 34.1 | 27.7 | 40.4 | 16 | 48 |
| `fused_rms_gated_fp8_group_quant` |  512 | 7168 | 8192 | main |   —  |   —  |   —  |  — |  — |
| `fused_rms_gated_fp8_group_quant` |  512 | 7168 | 8192 | PR   | 35.7 | 29.9 | 41.5 | 16 | 48 |
| `fused_rms_gated_fp8_group_quant` | 4096 | 7168 | 8192 | main |   —  |   —  |   —  |  — |  — |
| `fused_rms_gated_fp8_group_quant` | 4096 | 7168 | 8192 | PR   | 103.9 | 103.3 | 104.5 | 28 | 48 |
| `fused_rms_gated_fp8_group_quant` |   64 | 4096 | 4096 | main |   —  |   —  |   —  |  — |  — |
| `fused_rms_gated_fp8_group_quant` |   64 | 4096 | 4096 | PR   | 23.4 | 21.6 | 25.2 | 16 | 48 |

---

**Comment by Boss2002n** [comment]
The formatting - LGTM 😄

**Review by k50112113 — DISMISSED** [review]
No regression Great! Looks good!

**Comment by ChuanLi1101** [comment]
@hellozhuo-amd quick triage ask -- this PR currently has 3 real CI failures (not flakes; they ran for the full duration):

- `Flash Attention - Triton / MI35X (1 GPU)` -- 32m56s fail
- `Flash Attention - Triton / RDNA3 (1 GPU)` -- 40s fail (looks like an early init failure)
- `Triton Tests (MI35X) / Shard 3` -- 20m27s fail

The Flash Attention failures are unexpected since this PR doesn't touch attention paths. Two suspects:

1. Broken import chain from the `conv/` reorganization. The shim in `aiter/ops/triton/__init__.py` resolves `aiter.ops.triton.causal_conv1d` lazily, but a module-level `from aiter.ops.triton._triton_kernels.causal_conv1d import ...` in an attention test/dep would not resolve after the move under `_triton_kernels/conv/`.
2. The merged `_fused_rms_fp8_group_quant_kernel` constexpr count nearly doubled (added ~15 new `constexpr` args). If any attention test indirectly imports / autoload-touches this kernel, Triton's compile cache or a constexpr-corner-case could be the trigger.

Also, @Boss2002n's CHANGES_REQUESTED from 5/8 is still outstanding and the reviewDecision is still blocked on it. Could you push a triage commit + ping Boss2002n to resolve so we can re-review?

cc @maeehart @azaidy

**Comment by brunomazzottiamd** [comment]
@ChuanLi1101, regarding the **Flash Attention** job failures you mentioned in a previous comment:

> ...this PR currently has 3 real CI failures (not flakes; they ran for the full duration):
> 
> * `Flash Attention - Triton / MI35X (1 GPU)` — 32m56s fail
> * `Flash Attention - Triton / RDNA3 (1 GPU)` — 40s fail (looks like an early init failure)
> * `Triton Tests (MI35X) / Shard 3` — 20m27s fail

They are addressed by https://github.com/ROCm/aiter/pull/2695. IMHO they aren't a blocker for this PR.

**Review by Boss2002n — DISMISSED** [review]
LGTM! :)

**Comment by hellozhuo-amd** [comment]
@copilot resolve the merge conflicts in this pull request

**Comment by hellozhuo-amd** [comment]
> @ChuanLi1101, regarding the **Flash Attention** job failures you mentioned in a previous comment:
> 
> > ...this PR currently has 3 real CI failures (not flakes; they ran for the full duration):
> > 
> > * `Flash Attention - Triton / MI35X (1 GPU)` — 32m56s fail
> > * `Flash Attention - Triton / RDNA3 (1 GPU)` — 40s fail (looks like an early init failure)
> > * `Triton Tests (MI35X) / Shard 3` — 20m27s fail
> 
> They are addressed by #2695. IMHO they aren't a blocker for this PR.

@ChuanLi1101 Regarding this, what's your opinion?

**Comment by hellozhuo-amd** [comment]
@copilot resolve the merge conflicts in this pull request

**Review by k50112113 — APPROVED** [review]
LGTM
